clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -909,6 +880,11 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     stream_status_t
     eof_or_idle(output_ordinal_t output, input_ordinal_t prev_input);
 
+    // mapping_t-mode specific actions when one output runs out of things to do.
```

Comment:
How does this relate to pick_next_input_for_mode? When one output runs out of things to do, one might expect the next input to be picked (using pick_next_input_for_mode) instead of going idle or just ending (via eof).

Reply:
This is when pick_next_input_for_mode() has run out of things to do so the only source of more work would be stealing/rebalancing from some other output, and whether this output should wait for that is mode-dependent.  Changed the comment to:

    // mapping_t-mode specific actions when one output runs out of things to do
    // (pick_next_input_for_mode() has nothing left in this output's queue).
    // Success return values are either STATUS_IDLE (asking the user to keep
    // polling as more work may show up in the future) or STATUS_EOF.